### PR TITLE
Added GPS information and 

### PR DESCRIPTION
**Pull request by fabcira**
Main Changes 
-------------------
- introduced GPS information
- introduced some of the missing connections between the preferences and the actual code
- modified the way data is uploaded to the server:
     1. long/lat and time are not associated to every single reading. This is a waste of data upload. Data is compacted into a list containing all the variables and which is uploaded in one go with a common  time/lat/long.
     2. taken care of cases where the server is not responding: created a local database where data is temporarily stored. Data will be sent next time the app is loaded
- the interface does not have a table with all data but a textview with only the current data: this is much easier to read
- the font color has been changed to greyish to white (I could not read it)
- the names of the views in the interface have now a positional name (row1_column2). This is because in the future it would be great to have the allocation of data flexible (so that you can change what is displayed where
- other minor changes

Issues
--------
The data displayed is wrong. This is not an issue with this branch. It is something that happens also in the original repository.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,6 +29,20 @@ If you want to upload data to a server, for now, check the following:
...
+common structure and then it is uploaded in one go
+* taken care of cases where the server is not responding: created a local database where data is temporarily stored
+* the interface does not have a table with all data but a textview with only the current data: this is much easier to read 
+* the font color has been changed to greyish to white (I could not read it)
+* the names of the views in the interface have now a positional name (row1_column2). This is because in the future it would be great to have the allocation of data flexible (so that you can change what is displayed where
+* other minor changes
+
+
+## Issues
+The data displayed is wrong. This is not an issue with this branch. It is something that happens also in the original repository.
```
Comment (pires) [inline README.md]: Ah, interesting.

File: src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -14,9 +14,14 @@
...
+    //added FC
+    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"/>
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
+
+
   <application
     android:allowBackup="true"
-    android:icon="@drawable/ic_launcher"
+      android:icon="@drawable/elm327e"
     android:label="@string/app_name"
```
Comment (pires) [inline src/main/AndroidManifest.xml]: Sure!

**Comment by pires** [comment]
@fabcira want to let you know this contribution is of value to me. Unfortunately, it's really hard to verify such a huge patch. Hope you have the time and will to catch up with the current code and clean-up this PR ;-)

**Comment by nomwerp** [comment]
closing since by now we've picked all the good bits or implemented them otherwise , thanks again!

**Pull request closed without merging**